client/src/main/java/uk/co/blackpepper/bowman/LinkedResourceMethodHandler.java

```diff
@@ -82,49 +82,64 @@ private Object invokeAnnotatedMethod(Object self, Method method, Method proceed,
 	}
 
 	private Object resolveLinkedResource(Object self, Method method, Method proceed, Object[] args)
-			throws IllegalAccessException, InvocationTargetException {
-		
+	throws IllegalAccessException, InvocationTargetException {
+
 		URI associationResource = new MethodLinkUriResolver(resource).resolveForMethod(method, args);
-		
+
 		if (Collection.class.isAssignableFrom(method.getReturnType())) {
 			Class<?> linkedEntityType = (Class<?>) ((ParameterizedType) method.getGenericReturnType())
-				.getActualTypeArguments()[0];
-			
-			return resolveCollectionLinkedResource(associationResource, linkedEntityType, self, proceed);
-		}
+					.getActualTypeArguments()[0];
 
+			if (proceed == null) {
+				return createCollectionOfLinkedResource(getLinkedResources(associationResource,
+						linkedEntityType), method);
+			}
+			else {
+				return createCollectionOfLinkedResource(getLinkedResources(associationResource, linkedEntityType),
+						self, proceed);
+			}
+		}
 		return resolveSingleLinkedResource(associationResource, method.getReturnType());
 	}
 
 	private <F> F resolveSingleLinkedResource(URI associationResource, Class<F> linkedEntityType) {
 		Resource<F> linkedResource = restOperations.getResource(associationResource, linkedEntityType);
-		
+
 		if (linkedResource == null) {
 			return null;
 		}
-		
+
 		return proxyFactory.create(linkedResource, restOperations);
 	}
 
-	private <F> Collection<F> resolveCollectionLinkedResource(URI associationResource, Class<F> linkedEntityType,
-		Object contextEntity, Method originalMethod) throws IllegalAccessException, InvocationTargetException {
-		
-		Resources<Resource<F>> resources = restOperations.getResources(associationResource, linkedEntityType);
-		
-		@SuppressWarnings("unchecked")
+	private <F> Collection<F> createCollectionOfLinkedResource(Resources<Resource<F>> resources, Object contextEntity,
+		Method originalMethod) throws IllegalAccessException, InvocationTargetException {
+
+		// noinspection unchecked
```

Comment:
Please use `@SuppressWarnings("unchecked")` to stop the compiler warnings rather than IDE-specific comments - I know this is a pain as you can't put this on an assignment, so you'll need an intermediate local variable declaration to put this on. Still I think that's better than putting IDEA-specific stuff in the code.